data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/consumer/MultithreadedConsumerTest.java

```diff
@@ -0,0 +1,126 @@
+        Yaml yaml = new Yaml();
+        FileReader fileReader = new FileReader(getClass().getClassLoader().getResource("sample-pipelines.yaml").getFile());
+        Object data = yaml.load(fileReader);
+        if (data instanceof Map) {
+            Map<String, Object> propertyMap = (Map<String, Object>) data;
+            Map<String, Object> logPipelineMap = (Map<String, Object>) propertyMap.get("log-pipeline");
+            Map<String, Object> sourceMap = (Map<String, Object>) logPipelineMap.get("source");
+            Map<String, Object> kafkaConfigMap = (Map<String, Object>) sourceMap.get("kafka");
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.registerModule(new JavaTimeModule());
+            String json = mapper.writeValueAsString(kafkaConfigMap);
+            Reader reader = new StringReader(json);
+            sourceConfig = mapper.readValue(reader, KafkaSourceConfig.class);
+            topicConfig = sourceConfig.getTopics().get(0);
+        }
+        pluginMetrics = mock(PluginMetrics.class);
+        buffer = mock(Buffer.class);
+    }
+
+    private MultithreadedConsumer createObjectUnderTest(String consumerId,
+                                                        String consumerGroupId,
+                                                        String schema){
+        properties = new Properties();
+        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVERS);
+        properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        return new MultithreadedConsumer(consumerId,
+                consumerGroupId,
+                properties,
+                topicConfig,
+                sourceConfig,
+                buffer,
+                pluginMetrics,
+                schema);
+    }
+
+   // @ParameterizedTest
+   // @ValueSource(strings = "plaintext")
+    @Test
+    void testRunWithPlainText() throws InterruptedException {
+        String topic = topicConfig.getName();
+        //schemaType = "plaintext";
+        Map<TopicPartition, List<ConsumerRecord<String, Object>>> records = new LinkedHashMap<>();
+        Thread producerThread = new Thread(() -> {
+            ConsumerRecord<String, Object> record1 = new ConsumerRecord<>(topic, 0, 0L, "mykey-1", "myvalue-1");
+            ConsumerRecord<String, Object> record2 = new ConsumerRecord<>(topic, 0, 0L, "mykey-2", "myvalue-2");
+            records.put(new TopicPartition(topic, 1), Arrays.asList(record1, record2));
+        });
+        producerThread.start();
+        TimeUnit.SECONDS.sleep(1);
+        producerThread.join();
+        multithreadedConsumer = createObjectUnderTest("DPKafkaProj-1",
+                "DPKafkaProj-1","plaintext");
+        MultithreadedConsumer spySource = spy(multithreadedConsumer);
+        doCallRealMethod().when(spySource).run();
```

Comment:
Need to test the functionality of the code.

Reply:
We are working on this Junit and will update the same in this PR or PR#2713